backend/src/main/java/eu/clarin/switchboard/resources/DataResource.java

```diff
@@ -52,11 +54,44 @@ public Response getFile(@PathParam("id") String idString)  {
         };
 
         Response.ResponseBuilder builder = Response.ok(fileStream);
-        builder.type(fi.getProfile().getMediaType());
+        builder.type(fi.getProfile().toProfile().getMediaType());
         builder.header("content-disposition", "attachment; filename=" + fi.getFilename());
         return builder.build();
     }
 
+    @GET
+    @Path("/{id}/info")
+    @Produces(MediaType.APPLICATION_JSON + ";charset=utf-8")
+    public Response getFileInfo(@Context HttpServletRequest request, @PathParam("id") String idString)
+            throws Exception {
+        UUID id;
+        try {
+            id = UUID.fromString(idString);
+        } catch (IllegalArgumentException xc) {
+            return Response.status(Response.Status.NOT_FOUND).build();
+        }
+
+        while (true) {
```

Comment:
Nob question (my Java is very rusty): can't this loop extremely fast on line 88 and consume a lot of CPU?

Reply:
Yes of course, thanks for catching this but. And actually we don't need a loop but just a future, I will rewrite this. The other comments are valid too.